Maple2.Model/Metadata/MapEntity/FieldEntity.cs

```diff
@@ -0,0 +1,85 @@
+﻿using Maple2.Tools.VectorMath;
+using System.Numerics;
+
+namespace Maple2.Model.Metadata.FieldEntities;
+
+public enum FieldEntityType : byte {
+    Unknown,
+    Vibrate,
+    SpawnTile, // for mob spawns
+    BoxCollider, // for cube tiles (IsWhiteBox = false) & arbitrarily sized white boxes
+    MeshCollider,
+    Fluid,
+    Cell // for cells culled from the grid & demoted to AABB tree
+}
+
+public record FieldEntityId(
+    ulong High,
+    ulong Low) {
+    public bool IsNull { get => High == 0 && Low == 0; }
+
+    public static FieldEntityId FromString(string id) {
+        ulong High = Convert.ToUInt64(id.Substring(0, 16), 16);
+        ulong Low = Convert.ToUInt64(id.Substring(16, 16), 16);
+
+        return new FieldEntityId(High, Low);
+    }
+}
+
+public record FieldEntity(
```

Comment:
It's specialized to how FieldAccelerationStructure needs it, because it's specialized in spatial queries. It needs a specific format and data layout to load correctly. It was already going to be stored in a different way than MapEntity objects were, and it has different needs than what MapEntity set out to accomplish, so it didn't make sense to use the same interface.

Reply:
But what makes this different that you can't just combine it both, it'll just contain additional fields required for FieldAccelerationStructure. My problem with this is that we're building the same boat twice. 
If I wanted to get blocks around me and figure out each type, properties, etc, I would query it, get the block ID, then search DB once more using that block ID. It's a cost of two look ups.

Reply:
It doesn't store the same entities as what is in MapEntity. Any data relating to entities included in this should probably be added to this rather than the other, so there wouldn't be 2 look ups. 

The XBlock field in the other one appears to have been added to help SQL queries identify which entities belong to which map, but it isn't necessary for this because they're all stored in the same serialized byte array for the same map.

As for look up cost of this vs look up cost in MapEntity, the FieldAccelerationStructure query functions aren't DB queries. They're in memory queries that minimize the # of objects checked to reduce costs. In most cases the queries should be barely more than doing bounding box intersection checks on immediately neighboring objects, if even that much.  It's also optimized for cache friendliness so the footprint should be insignificantly small for anything doing less than a full scene physics update or frame render.

Reply:
I'm not saying one is better than the other. My point is that if this is the preferred usage, more optimal, and cache friendly of MapEntities/FieldMapEntities, why don't we just retire MapEntity and use this instead? For the sake of ease usage.

Reply:
I'm not either. I'm saying that it had different needs because it fulfilled a different purpose from what MapEntity looked like it was used for. MapEntity has a lot of queries that look use case specific, like looking up by trigger name. We could if you want to, but if the entities don't need spatial queries then adding them to the FieldAccelerationStructure will slow it down for the queries that do need that type of query.

The grid aligned ones wouldn't add too much overhead outside of big queries, but any extra unaligned ones that are added will need to be added to the AABB tree, making it deeper than it needs to be. The deeper the AABB tree is, and the more objects are clustered in certain spots inside the tree, the slower the queries will be for objects that do need spatial queries.